File: 0933-number-of-recent-calls/0933-number-of-recent-calls.py
from collections import deque


class RecentCounter:
    def __init__(self):
        self.que = deque()


    def ping(self, t: int) -> int:
        while self.que and t - self.que[0] > 3000:
            self.que.popleft()
        self.que.append(t)
        return len(self.que)

File: 0933-number-of-recent-calls/0933-number-of-recent-calls.java
class RecentCounter {
    Queue<Integer> que;

    public RecentCounter() {
        que = new ArrayDeque<>();
    }
    
    public int ping(int t) {
        while (!que.isEmpty() && t - que.peek() > 3000) {
            que.poll();
        }
        que.add(t);
        return que.size();
    }
}
